### ISSUE TITLE
Steam auto-logs out and quit my game without giving a chance to save.

### ISSUE THREAD

**Issue by leklachu** [ValveSoftware/steam-for-linux#9255]
#### Your system information

* Steam client version (build number or date): 2023-02-14, 1676336721
* Distribution (e.g. Ubuntu): OpenSUSE Tumbleweed
* Opted into Steam client beta?: [Yes/No] no
* Have you checked for system updates?: [Yes/No] unable currently

#### Please describe your issue in as much detail as possible:
Describe what you _expected_ should happen and what _did_ happen. Please link any large code pastes as a [Github Gist](https://gist.github.com/)

Steam has been struggling with my internet connection: not sure whose fault it is. My net connection seems to be fine, but could be to blame for the connection.

Twice now Steam has failed to log in, and asked me to sign in again. When I do so, it claims I've logged in too many times, please try again later. Starting offline still works. Starting online also still works (at least after the previous time this happened) later once - presumably - the network connection is stable again, and requires no extra login.

That makes decent enough sense with some networking problem confusing things, though it still would be nice not to fail like that. The real problem is, this second time it happened while in a game. (First time was when starting Steam, or perhaps waking from suspend.) Having failed the re-login for "too many tries" (on my first try!), Steam auto-quit, and quit the game without offering a chance to save!

tl;dr: Steam had connection trouble and couldn't "refresh" its login, so it quit without opportunity to save my game.

#### Steps for reproducing this issue:

1.  Have strange networking issues
2. Wait for Steam to require a re-login 
3. Correctly type password, and watch while Steam decides to quit anyway.

**Comment by leklachu** [comment]
Addendum: Yep, now a bit later it opens fine in online mode, no "refresh login" or anything required.
